src/Core/Managed/Shared/Extensibility/MetricManager.cs

```diff
@@ -0,0 +1,285 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+#if CORE_PCL || NET45 || NET46
+    using TaskEx = System.Threading.Tasks.Task;
+#endif
+
+    /// <summary>
+    /// Metric factory and controller.
+    /// </summary>
+    public sealed class MetricManager : IDisposable
+    {
+        /// <summary>
+        /// Name of the property added to aggregation results to indicate duration of the aggregation interval.
+        /// </summary>
+        private static string intervalDurationPropertyName = "IntervalDurationMs";
+
+        /// <summary>
+        /// Reporting frequency.
+        /// </summary>
+        private static TimeSpan snapshotFrequency = TimeSpan.FromMinutes(1);
```

Comment:
aggregationPeriod would be a better name. frequency is a *rate*, but you are using a time span.